python/langsmith/evaluation/_runner.py

```diff
@@ -1678,12 +1688,31 @@ def _ensure_traceable(
             "    ...\n"
             ")"
         )
+
     if rh.is_traceable_function(target):
         fn = target
     else:
         fn = rh.traceable(name="Target")(target)
     return fn
 
+def _with_attachments(
+    target: TARGET_T | rh.SupportsLangsmithExtra[[dict], dict],       
+) -> bool:
+    """Whether the target function accepts attachments."""
+    # Check function signature
+    sig = inspect.signature(target)
+    params = list(sig.parameters.values())
+    positional_params = [p for p in params if p.kind in (p.POSITIONAL_ONLY, p.POSITIONAL_OR_KEYWORD)]
+    
+    if len(positional_params) == 0:
+        raise ValueError("Target function must accept at least one positional argument (inputs)")
+    elif len(positional_params) > 2:
```

Comment:
@hinthornw @baskaryan just calling out that _currently_, an experiment still gets created but none of the runs succeed with >1 positional argument is supplied. so this isn't exactly a breaking change but will change behavior, I think for the better. LMK if you have strong opinions though